tests/queries/0_stateless/02984_variant_type_mutations.sh

```diff
@@ -0,0 +1,114 @@
+{
+    $CH_CLIENT -q "drop table if exists test;"
+}
+
+function insert_data()
+{
+  $CH_CLIENT -q  "insert into test select number, number from numbers(3);"
+}
+
+function update_data()
+{
+    data_type="UInt64"
+    $CH_CLIENT -nmq "select id, v.$data_type from test;"
+
+    # Update Variant data while keeping the existing data type
+    $CH_CLIENT -q  "alter table test update v = 1::$data_type where 1;"
+    $CH_CLIENT -nmq "select id, v.$data_type from test;"
+    # -----------------------------------------------------------------
+
+    # Update Variant data while changing to a different data type
+    data_type="Float64"
+    $CH_CLIENT -q  "alter table test update v = (1 - 0.9)::$data_type where 1;"
+    $CH_CLIENT -nmq "select id, v.\`$data_type\` from test order by id;"
+
+    data_type="Bool"
+    $CH_CLIENT -q  "alter table test update v = true::$data_type where 1;"
+    $CH_CLIENT -nmq "select id, v.\`$data_type\` from test order by id;"
+
+    data_type="DateTime"
+    $CH_CLIENT -q  "alter table test update v = '2019-01-01 00:00:00'::$data_type where 1;"
+    $CH_CLIENT -nmq "select id, v.\`$data_type\` from test order by id;"
+
+    data_type="UUID"
+    $CH_CLIENT -q  "alter table test update v = toUUID('61f0c404-5cb3-11e7-907b-a6006ad3dba0')::$data_type where 1;"
+    $CH_CLIENT -nmq "select id, v.\`$data_type\` from test order by id;"
+
+    data_type="Array(UInt64)"
+    $CH_CLIENT -q  "alter table test update v = [1,2,3,4,5]::$data_type where 1;"
+    $CH_CLIENT -nmq "select id, v.\`$data_type\` from test order by id;"
+
+    data_type="Map(String, UInt64)"
+    $CH_CLIENT -q  "alter table test update v = map('one', 1, 'two', 2, 'three', 3)::$data_type where 1;"
+    $CH_CLIENT -nmq "select id, v.\`$data_type\` from test order by id;"
+
+    data_type="Tuple(i UInt8, s String)"
+    $CH_CLIENT -q  "alter table test update v = (1, 'one')::$data_type where 1;"
+    $CH_CLIENT -nmq "select id, v.\`$data_type\` from test order by id;"
+
+    data_type="IPv4"
+    $CH_CLIENT -q  "alter table test update v = '116.253.40.133'::$data_type where 1;"
+    $CH_CLIENT -nmq "select id, v.\`$data_type\` from test order by id;"
+
+    data_type="Ring"
+    $CH_CLIENT -q  "alter table test update v = [(0, 0), (10, 0), (10, 10), (0, 10)]::$data_type where 1;"
+    $CH_CLIENT -nmq "select id, v.\`$data_type\` from test order by id;"
+
+    data_type="String"
+    $CH_CLIENT -q  "alter table test update v = 'Hello'::$data_type where 1;"
+    $CH_CLIENT -nmq "select id, v.\`$data_type\` from test order by id;"
+    # -----------------------------------------------------------------
+}
+
+function delete_data()
+{
+    echo "Delete Data"
+
+    # Deleting with a condition on non-Variant column
+    $CH_CLIENT -q  "alter table test delete where id = 2;"
+    $CH_CLIENT -nmq "select id, v from test order by id;"
+
+    # Delete with a condition on Variant column value
+    #$CH_CLIENT -q  "alter table test delete where v = 'Hello';"
```

Comment:
Hi @Avogar, It looks like Clickhouse server crashes when Variant column is involved in a where condition. Not 100% sure whether Variant column supports where conditions though.

Reply:
This is 100% a bug, good that you catched it. I will fix it, it should work.